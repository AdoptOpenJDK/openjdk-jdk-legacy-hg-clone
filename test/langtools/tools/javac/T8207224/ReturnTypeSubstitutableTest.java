/*
 * @test /nodynamiccopyright/
 * @bug 8207224
 * @summary Javac compiles source code despite illegal use of unchecked conversions
 * @compile/fail/ref=ReturnTypeSubstitutableTest.out -XDrawDiagnostics ReturnTypeSubstitutableTest.java
 * @compile -source 12 ReturnTypeSubstitutableTest.java
 */

class ReturnTypeSubstitutableTest {
    abstract class AbstractDemo<Request extends AbstractResult, Response extends AbstractResult> {
        protected abstract Response test(Request request);
    }

    abstract interface AbstractResult {}

    abstract interface SimpleResult extends AbstractResult {}

    class Result1 implements SimpleResult {}

    class OtherResult implements AbstractResult {}

    public class SimpleDemo<Request extends AbstractResult, Response extends AbstractResult> extends AbstractDemo<Request, Response> {
        @Override
        protected SimpleResult test(AbstractResult request) {
            return null;
        }
    }
}
