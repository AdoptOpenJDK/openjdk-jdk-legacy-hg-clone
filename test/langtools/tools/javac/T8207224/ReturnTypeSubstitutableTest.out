ReturnTypeSubstitutableTest.java:24:32: compiler.err.override.incompatible.ret: (compiler.misc.cant.override: test(ReturnTypeSubstitutableTest.AbstractResult), ReturnTypeSubstitutableTest.SimpleDemo, test(Request), ReturnTypeSubstitutableTest.AbstractDemo), ReturnTypeSubstitutableTest.SimpleResult, Response
1 error
